sites/main-site/src/content/blog/2025/modules-ontology.mdx

```diff
@@ -0,0 +1,324 @@
+                    type: file
+                    description: Optional index file for BAM file
+                    pattern: "*.{csi}"
+            - crai:
+                - meta:
+                    type: file
+                    description: Groovy Map containing sample information
+                - "*.crai":
+                    type: file
+                    description: Optional index file for CRAM file
+                    pattern: "*.{crai}"
+            - versions:
+                - versions.yml:
+                    type: file
+                    description: File containing software versions
+                    pattern: "versions.yml"
+        ...
+        ```
+    </div>
+
+</div>
+
+We also introduced linting checks to nf-core/tools to ensure the proper structure of `meta.yml` files.
+
+Together with these linting checks, we also introduced a new flag to the `nf-core modules lint` command: `--fix`.
+This flag will try to fix all the possible lint failures related to the `meta.yml` file.
+
+### Introducing ontologies
+
+Together with these changes, we also added the [`bio.tools`](https://bio.tools/) identifier of the tool.
+
+```yml title="meta.yml" {13}
+name: bwa_mem
+
+    ...
+
+tools:
+    - bwa:
+          description: |
+              BWA is a software package for mapping DNA
+              sequences against a large reference genome, such as the human genome. homepage: http://bio-bwa.sourceforge.net/
+          documentation: https://bio-bwa.sourceforge.net/bwa.shtml
+          arxiv: arXiv:1303.3997
+          licence: ["GPL-3.0-or-later"]
+          identifier: "biotools:bwa"
+```
+
+This bio.tools ID opened new possibilities, such as being able to know the inputs and outputs of the tool and the ontology terms for input and output files.
+
+We now use this information to generate a suggestion of input and output channels when creating a module with `nf-core modules create`
+(available as of February 2025 on the `dev` version of nf-core/tools and it will be released as part of nf-core/tools 3.3.0)
+So modifying the freshly created template of a module will be easier.
+
+Here we show an example of the template generated when creating a module for the BWA tool, using the same example as before.
+
+```yml title="main.nf"
+process BWA_MEM {
+    tag "$meta.id"
+    label 'process_single'
+
+    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bwa:0.7.18--h577a1d6_2':
+        'biocontainers/bwa:0.7.18--h577a1d6_2' }"
+
+    input:
+    // TODO nf-core: Update the information obtained from bio.tools and make sure that it is correct
+    tuple val(meta), path(sequence)
+    tuple val(meta), path(genome_index)
```

Comment:
Should the meta numbering increase with each input channel?

Reply:
yes, good catch!